src/instructlab/model/serve.py

```diff
@@ -121,6 +129,14 @@ def serve(
     from instructlab.model.backends import llama_cpp, vllm
 
     host, port = utils.split_hostport(ctx.obj.config.serve.host_port)
+
+    if gpus is not None:
```

Comment:
The logic I had https://github.com/instructlab/instructlab/pull/1974/files seems tighter.

There are two collisions you need to resolve/add logging for:
1) If tensor parallel size is set in vllm_args in the config. My PR proposes using the flag and throw a warning about it.
2) If tensor parallel size is set in ctx.args. This PR and mine propsed use ctx.args.

It looks like you're just missing 1) in this PR.

My PR also moves this logic down into the vLLM section and uses `contains_argument()` as Dan suggested.